Jenkinsfile

```diff
@@ -27,18 +27,51 @@ pipeline {
         script {
           setupGPGEnvironment()
           withCredentials([string(credentialsId: 'gpg.password', variable: 'GPG_PWD')]) {
-            def phase = isReleaseOrMasterBranch() ? 'deploy site-deploy' : 'verify'
+            def phase = isReleaseOrMasterBranch() ? 'deploy' : 'verify'
             maven cmd: "clean ${phase} " +
               "-Dgpg.skip=false " +
               "-Dgpg.project-build.password='${env.GPG_PWD}' " +
-              "-Dgithub.site.skip=${params.skipGitHubSite} " +
               "-Divy.engine.list.url=${params.engineListUrl} " +
               "-Dmaven.test.failure.ignore=true"
           }
           collectBuildArtifacts()
         }
       }
     }
+    stage('deploy-site') {
+      when {
+        expression { !params.skipGitHubSite && isReleaseOrMasterBranch() }
+      }
+      steps {
+        script {
+          maven cmd: "clean site " +
+            "-Divy.engine.list.url=${params.engineListUrl} "
+
+          withEnv(['GIT_SSH_COMMAND=ssh -o StrictHostKeyChecking=no']) {
+            sshagent(credentials: ['github-axonivy']) {
+              def branch = "pages_$BRANCH_NAME-$BUILD_NUMBER"
+              def message = "Update site (${env.BRANCH_NAME})"
+              sh """ 
+                git config --global user.name 'ivy-team'
+                git config --global user.email 'info@ivyteam.ch'
+
+                git clone -b gh-pages git@github.com:axonivy/project-build-plugin.git target/site-deployer
+                cd target/site-deployer
+                git checkout -b ${branch}
+                cd ../site
+                cp -r -v . ../site-deployer
+                cd ../site-deployer

```

Comment:
nasty copying could be avoided by:

1. cloning the branch into target/site
2. building with maven: enriches this target/site dir
3. then again in git env: commit/push

would you prefer this?